### PR TITLE
Improved alternate key pattern content.

### PR DESCRIPTION
**Pull request by gareth-a-jones**
Updated the content for clarity, directness, minor corrections on the scope of key uniqueness, fuller error examples, and being specific with guidance against multi-part alternate keys.
Also introduced the concept of key parentheses for those Graph users who've never encountered them, given slashes are the known public format.

### CODE REVIEW

File: graph/patterns/alternate-key.md [hunk 1]
```diff
@@ -2,29 +2,51 @@
 
 Microsoft Graph API Design Pattern
 
-*The alternate key pattern provides the ability to query for a single, specific resource identifiable through an alternative set of properties that is not its primary key.*
+*The alternate key pattern provides the ability to query for a single, specific resource identifiable via one of an alternative set of properties that is not its primary key.*
 
 ## Problem
 
-The resources exposed in Microsoft Graph are identified through a primary key, which guarantees uniqueness inside the same resource type. Often though, that same resource can also be uniquely identified by an alternative, more convenient property (or set of properties) that provides a better developer experience.
+The resources exposed in Microsoft Graph are identified through a primary key, which guarantees uniqueness inside the same resource collection. Often though, that same resource can also be uniquely identified by an alternative, more convenient property that provides a better developer experience.
```
Comment (XVincentX) [inline graph/patterns/alternate-key.md:9]: I am assuming composite keys are not a thing in MS Graph?

File: graph/patterns/alternate-key.md [hunk 2]
```diff
@@ -2,29 +2,51 @@
...
 
+```http
 https://graph.microsoft.com/v1.0/users(email='bob@contoso.com') Retrieves the employee with the email matching `bob@contoso.com`.
+```
+
+In the same way as requesting a resource via the canonical key, if a resource cannot be located that matches the alternate key, then a 404 must be returned.
+
+> **Note:** When requesting a resource via alternate keys, the simple slash-delimited URL style does not work.
+
+> **Note:** Do not use multi-part alternate keys.   Feedback has been that customers find multi-part keys confusing.
```
Comment (XVincentX) [inline graph/patterns/alternate-key.md:44]: What is a multi part key?

**Review by XVincentX — APPROVED** [review]
Great improvements, thanks!

File: graph/patterns/alternate-key.md (hunk 1 above)
Comment (gareth-a-jones) [reply graph/patterns/alternate-key.md:9]: correct

File: graph/patterns/alternate-key.md (hunk 2 above)
Comment (gareth-a-jones) [reply graph/patterns/alternate-key.md:44]: composite key

**Pull request merged**